### PR TITLE
feat: new average method length metric add

### PR DESCRIPTION
**Pull request by veledara**
This pull request adds a new metric, Average Method Length (AML), to the CAM project. The metric calculates the average number of lines per method in a class.

### CODE REVIEW

**Comment by veledara** [comment]
@yegor256 could you check, please?

File: metrics/aml.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        length = end_line - start_line + 1
+        total_length += length
+        count += 1
+    return total_length / count if count else 0.0
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print(
+            "Usage: python average_method_length.py <path to the .java file> <output file with metrics>"
```
Comment (yegor256) [inline metrics/aml.py:58]: @veledara maybe a typo? the script name is `aml.py`

File: tests/metrics/test-aml.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        System.out.println("Line 3");
+    }
+}
+EOT
+
+    metrics_file="${temp}/metrics.txt"
+    "${script_location}" "${java_file}" "${metrics_file}"
+    output=$(cat "${metrics_file}")
+    echo "${output}"
+    echo "${output}" | grep "AML" || { echo "Error: AML metric not found"; exit 1; }
```
Comment (yegor256) [inline tests/metrics/test-aml.sh:29]: @veledara maybe it's good to check for the actual value of the metric being calculated?

**Comment by yegor256** [comment]
@veledara in general, it's not a good idea to just submit pull requests with a new feature. First, we should submit a ticket and make sure it's accepted by the team. Then, we make a pull request. Otherwise, it's a big risk that your PR will simply be rejected as "not needed".

**Comment by veledara** [comment]
@yegor256 done, could you check, please?

File: tests/metrics/test-aml.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        System.out.println("Line 1");
+        System.out.println("Line 2");
+        System.out.println("Line 3");
+    }
+}
+EOT
+    metrics_file="${temp}/metrics.txt"
+    "${script_location}" "${java_file}" "${metrics_file}"
+    output=$(cat "${metrics_file}")
+    echo "${output}"
```
Comment (yegor256) [inline tests/metrics/test-aml.sh:26]: @veledara what this line is for?

Comment (veledara) [reply tests/metrics/test-aml.sh]: @yegor256 this line is redundant, I deleted it.

**Comment by veledara** [comment]
@yegor256 it is all done, could you check, please?

File: tests/metrics/test-aml.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        System.out.println("Line 1");
+        System.out.println("Line 2");
+        System.out.println("Line 3");
+    }
+}
+EOT
+    metrics_file="${temp}/metrics.txt"
+    "${script_location}" "${java_file}" "${metrics_file}"
+    output=$(cat "${metrics_file}")
+    expected_value="2.5"
```
Comment (yegor256) [inline tests/metrics/test-aml.sh:26]: @veledara why the value is 2.5? We have two methods here. One is 1 line, another is 5 lines. The average should 3, I believe?

File: tests/metrics/test-aml.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    echo "${output}" | grep -q "AML ${expected_value} " || {
+        echo "Error: Expected AML ${expected_value} not found in output:";
+        echo "${output}";
+        exit 1;
+    }
+    echo "${output}" | grep -q "Average Method Length" || {
+        echo "Error: Metric description 'Average Method Length' not found";
+        exit 1;
+    }
+} > "${stdout}" 2>&1
```
Comment (yegor256) [inline tests/metrics/test-aml.sh:36]: @veledara It would be beneficial to add one more test for a class with no methods, to see AML equal to zero in the end.

**Comment by veledara** [comment]
@yegor256 it is all done, could you check, please?

File: tests/metrics/test-aml.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    echo "${output}" | grep -q "AML ${expected_value} " || {
+        echo "Error: Expected AML ${expected_value} not found in output for empty class:";
+        echo "${output}";
+        exit 1;
+    }
+    echo "${output}" | grep -q "Average Method Length" || {
+        echo "Error: Metric description 'Average Method Length' not found in output for empty class";
+        exit 1;
+    }
+    echo "👍🏻 Test 2: Average Method Length was calculated correctly for empty class with value ${expected_value}"
```
Comment (yegor256) [inline tests/metrics/test-aml.sh:58]: @veledara what's the point of using this "Test N" notation? We don't do this in other tests. Be careful with ChatGPT, check what it does.

**Comment by veledara** [comment]
> @veledara what's the point of using this "Test N" notation? We don't do this in other tests. Be careful with ChatGPT, check what it does.

@yegor256 Okay, I removed this notation. But can you explain me, what was the problem? I swear that I wrote this addon in echo by myself (no ai tools were used) just to clarify that now we have two tests in the file

**Comment by yegor256** [comment]
@veledara we have another script called `ast.py`. It also scans Java files using the same Python library. In this script we keep all metrics that need AST information. For the sake of higher performance, it's better to put your code there as well. Otherwise, we'll do scan Java files twice, which will negatively affect the performance of the entire project.

**Comment by veledara** [comment]
@yegor256 done, could you check, please?

**Comment by yegor256** [comment]
@veledara thanks!
